bake/bake.go

```diff
@@ -0,0 +1,344 @@
+		case "platform":
+			t.Platforms = append(t.Platforms, parts[1])
+		default:
+			return errors.Errorf("unknown key: %s", keys[1])
+		}
+		c.Target[name] = t
+	}
+	return nil
+}
+
+func (c Config) ResolveGroup(name string) []string {
+	return c.group(name, map[string]struct{}{})
+}
+
+func (c Config) group(name string, visited map[string]struct{}) []string {
+	if _, ok := visited[name]; ok {
+		return nil
+	}
+	g, ok := c.Group[name]
+	if !ok {
+		return []string{name}
+	}
+	visited[name] = struct{}{}
+	targets := make([]string, 0, len(g.Targets))
+	for _, t := range g.Targets {
+		targets = append(targets, c.group(t, visited)...)
+	}
+	return targets
+}
+
+func (c Config) ResolveTarget(name string) (*Target, error) {
+	return c.target(name, map[string]struct{}{})
+}
+
+func (c Config) target(name string, visited map[string]struct{}) (*Target, error) {
+	if _, ok := visited[name]; ok {
+		return nil, nil
+	}
+	visited[name] = struct{}{}
+	t, ok := c.Target[name]
+	if !ok {
+		return nil, errors.Errorf("failed to find target %s", name)
+	}
+	var tt Target
+	for _, name := range t.Inherits {
+		t, err := c.target(name, visited)
+		if err != nil {
+			return nil, err
+		}
+		if t != nil {
+			tt = merge(tt, *t)
+		}
+	}
+	t.Inherits = nil
+	tt = merge(merge(defaultTarget(), t), tt)
+	tt.normalize()
+	return &tt, nil
+}
+
+type Group struct {
+	Targets []string
+	// Target // TODO?
+}
+
+type Target struct {
+	Inherits   []string          `json:"inherits,omitempty"`
+	Context    string            `json:"context,omitempty"`
+	Dockerfile string            `json:"dockerfile,omitempty"`
+	Args       map[string]string `json:"args,omitempty"`
+	Labels     map[string]string `json:"labels,omitempty"`
+	Tags       []string          `json:"tags,omitempty"`
+	CacheFrom  []string          `json:"cache-from,omitempty"`
+	Target     *string           `json:"target,omitempty"`
+	Secrets    []string          `json:"secret,omitempty"`
+	SSH        []string          `json:"ssh,omitempty"`
+	Platforms  []string          `json:"platform,omitempty"`
+}
+
+func (t *Target) normalize() {
+	t.Tags = removeDupes(t.Tags)
+	t.Secrets = removeDupes(t.Secrets)
+	t.SSH = removeDupes(t.SSH)
+	t.Platforms = removeDupes(t.Platforms)
+}
+
+func TargetsToBuildOpt(m map[string]Target) (map[string]build.Options, error) {
+	m2 := make(map[string]build.Options, len(m))
+	for k, v := range m {
+		bo, err := toBuildOpt(v)
+		if err != nil {
+			return nil, err
+		}
+		m2[k] = *bo
+	}
+	return m2, nil
+}
+
+func toBuildOpt(t Target) (*build.Options, error) {
+	if t.Context == "-" {
+		return nil, errors.Errorf("context from stdin not allowed in bake")
+	}
+	if t.Dockerfile == "-" {
+		return nil, errors.Errorf("dockerfile from stdin not allowed in bake")
+	}
+
+	bo := &build.Options{
+		Inputs: build.Inputs{
+			ContextPath:    t.Context,
+			DockerfilePath: t.Dockerfile,
+		},
+		Tags:      t.Tags,
+		BuildArgs: t.Args,
+		Labels:    t.Labels,
+		// CacheFrom: t.CacheFrom,
```

Comment:
?

Reply:
The key does not exist atm. but should later.